x-pack/plugins/apm/server/routes/metrics/by_agent/java/gc/fetch_and_transform_gc_metrics.ts

```diff
@@ -52,11 +58,26 @@ export async function fetchAndTransformGcMetrics({
   start: number;
   end: number;
   chartBase: ChartBase;
-  fieldName: typeof METRIC_JAVA_GC_COUNT | typeof METRIC_JAVA_GC_TIME;
+  rateOrTime: typeof RATE | typeof TIME;
   operationName: string;
+  isOpenTelemetry?: boolean;
 }) {
   const { bucketSize } = getBucketSize({ start, end });
 
+  const groupByField = isOpenTelemetry ? LABEL_GC : LABEL_NAME;
+
+  const targetField = isOpenTelemetry
+    ? METRIC_OTEL_JVM_GC_DURATION
+    : rateOrTime === RATE
+    ? METRIC_JAVA_GC_COUNT
+    : METRIC_JAVA_GC_TIME;
+
+  const fieldAggregation = isOpenTelemetry
+    ? rateOrTime === RATE
+      ? { value_count: { field: targetField } }
+      : { sum: { field: targetField } }
+    : { max: { field: targetField } };
```

Comment:
I wonder if `fetchAndTransformGcMetrics` should take `groupByField`, `targetField` and `fieldAggregation` as arguments instead of the conditionals here.

Reply:
Btw The `fetchAndTransform*Metrics` abstractions have annoyed me for years. Wondering if we'd be better of without them.